classes/class-extensions.php

```diff
@@ -0,0 +1,59 @@
+<?php
+
+namespace HM\BackUpWordPress;
+
+class Extensions {
+
+	private static $instance;
+
+	function __construct( $root_url ) {
+
+		$this->root_url  = $root_url;
+
+	}
+
+	public static function get_instance() {
+
+		if ( ! static::$instance ) {
+
+			$root_url = 'https://bwp.hmn.md/wp-json/wp/v2/';
+
+			static::$instance = new static( $root_url );
+		}
+
+		return static::$instance;
+	}
+
+	public function get_edd_data() {
+
+		$response = $this->fetch( 'edd-downloads', 600 );
+
+		if ( is_wp_error( $response ) || empty( $response['body'] ) ) {
+			return false;
+		}
+
+		return json_decode( $response['body'] );
+
+	}
+
+	function fetch( $endpoint, $ttl = 10 ) {
```

Comment:
Should we consider caching the data for much longer than this, say 24 hours? Otherwise the data will essentially refresh every time someone visits the page which could be slow.